### ISSUE TITLE
Is there a way to run  sickbeard_mp4_automator from remote PC

### ISSUE THREAD

**Issue by trentmsteel** [mdhiggins/sickbeard_mp4_automator#736]
I have been trying to figure out a way to the get the script to run on my Hyper-V host machine from, reason being the host machine is equipped with a GPU that is Intel Quick Sync compatible . 
My use case would be using Windows:

Sonnarr...etc ---> NZBget finishes downloading ---> Nzbget excutes sickbeard_mp4_automator on Hyper-V Host, when complete processing video using "h264qsv" ---> rest of process completes on Hyper-V VM as it normally would. 

I have tried a few things to no avail, and just wanted to know if doing this is currently out of the scope of this script at the moment.

Thanks for reading

**Comment by themichaelmitchell** [comment]
Just curious, what OS is sonarr running in? If it's a guest Windows OS, what if you make the script something like this:

`Invoke-Command hostcomputername-ScriptBlock {python "pathtoscript" -I "path to media" -tvdb "tvdb code"}`

That's off (I'm on mobile) but have you tested that way at all? Use the sonarr script and pass the variables using the Sonarr available variables maybe? I'll play with this when I get home

**Comment by trentmsteel** [comment]
Thanks for replying. It is indeed running on a Windows OS. As for what I have tried, I have tried installing sickbeard_mp4_automator on the Host OS which is also Windows, and just mounted the sickbeard_mp4_automator installation on on the Host into the Guest VM. 

Well needless to say that did not work, I have zero coding experience, but currently trying to learn Python. I will try your suggestion as soon as I figure out what it is referencing. Thanks again.

**Comment by justinglock40** [comment]
why not just run the script on VM?

**Comment by trentmsteel** [comment]
I did run it within the VM, the only draw back is that Hyper-V VM can not use the Intel Quick Sync feature of my On board GPU. And I find the Quick Sync Feature invaluable in the time it saves in time and CPU resources on my HTPC.

**Comment by justinglock40** [comment]
run the script on host OS then have Sonarr/Radarr/etc move the file back to a folder thats shared between the vm?

**Comment by trentmsteel** [comment]
I thought of that just yesterday after a week of trying everything else. I installed Sonarr, Radarr, etc on the VM. I then installed NZBGET as well as Sickbeard_mp4 on the Host and pointed the NZBGET  settings within Sonarr and Raddar to the ip address of NZBGET installed on the HOST.

 At the point everything seemed to work, NZBGET on the HOST downloaded then triggered sickbed_mp4's script which successfully processed the file. But an issue came up when Sonarr did not import the mp4 file that is now sitting in a folder on the HOST machine in a folder called "complete". This caused Sonarr to endlessly try to download the file.

I have tried changing all "localhost" found in the sickbeard_mp4 "autoProcess.ini" to the ip of the VM' to see if it would trigger Sonarr and Radarr to import, but to no affect. I have also tried changing all local paths found in Sonarr and Radarr to the network shared paths i.e. 
"\\HOSTNAME\". But in the end nothing got the processed file to be imported by Sonarr. At this point I am studying Python to get a grasp on the inner working of this awesome script and hopefully come up with a solution.

**Comment by justinglock40** [comment]
Did you try using the remote path option in sonarr?
<img width="1182" alt="screenshot at may 13 09-41-48" src="https://cloud.githubusercontent.com/assets/23133649/26026414/5c1a6354-37c0-11e7-8120-85aaa702b7d0.png">

**Comment by trentmsteel** [comment]
I did not. I'm going to give that a shot right now.

**Comment by trentmsteel** [comment]
Thanks for you suggestion it pointed me in the direction of positive results. Files are now being processed on the HOST and imported into the VM  NOTE: I tried the remote port mapping option, however Sonarr did not import the file from the HOST. 

However while trying Remote Path Mapping, it occurred to me to try the Drone Factory Options so with "Complete download Handling" disabled I inputed the network shared final destination of the file after sickbed_mp4 is done with it, (\\HOSTNAME\PATH\complete\sonarr) into Drone Factory. 

Also I reverted the settings in files on the HOST "autoProcess.ini" and NZBGET settings as if described in sickbeard_mp4's README. 
I will keep testing this setup, my concern are what will be if any adverse effects of using Drone Factory, as it seems Sonarr wiki advocates "Complete Download Handling" over the older Drone Factory.

**Comment by justinglock40** [comment]
If I’m right I think the drone factory is an issue because it scans at intervals. So if the file is still being process when it does its scan then it’ll move the file and case corruption. I think the Complete Downloads setting gets around this because it hooks in to the client and the client talks to Sonarr/Radarr and says ok I’m done with this. you can have it.

**Comment by trentmsteel** [comment]
I see. I also found [this post](https://www.reddit.com/r/sonarr/comments/44lasw/why_the_drone_factory_is_bad_or_just_plain_evil/) expanding on what you mentioned about Drone Factory not being the way to go about things. So it looks like it's back to the drawing board.

**Comment by justinglock40** [comment]
Is there any particular reason you can’t run everything on the Host?

**Comment by trentmsteel** [comment]
Well it comes down to Continuity of Service. I have for some time ran a personal Plex server for myself and family and friends. There are times when the Server becomes unreliable or even the occasional OS Patch Upgrades and oh yes system failures, failures don't happen often, but when it they do it may take minutes to hours to resolve and restore the server from system backups or doing it manually.

My thoughts are if I could run the Hyper-VM with Sonarr, Raddar, Plexpy, Nzbget..etc. And then enable replication with a setting as low as 30 seconds if need be on the Hyper-V Virtual Machine. I could deploy the replicated VMs to multiple VPS (at least 2) for less than $8.00 a month each running at two different VPS providers across the country. I don't think the VPS's needs to be all that powerful given all they would do is deliver the content already processed for steaming by Sickbed_mp4 on the original HOST using the "h264qsv" codec.

So in the event the Original HOST for what ever reason could not function, one of the VPS running a replicated VM would be activated by me and things would continue with no perceived black out, by the end user. At least that is my current perception on how things would work. 
(Note: All content would be stored by a cloud storage provider and accessed using Stablebit Cloud drive, which is installed on the VM and it's replicants.)

**Comment by trentmsteel** [comment]
Got it working

**Comment by justinglock40** [comment]
What did you end up having to do?

**Comment by trentmsteel** [comment]
I eventually suspended my efforts in trying to Macgyver the sickbed_mp4 code, given I am only now learning coding. And decided to look into seeing if Virtualizing a physical GPU within a Virtual Machine without a fancy GTX something or another, as not to bare the cost and complexity associated  with setting one of those up for a novice such as myself.  

So I looked into once again how to get a VM to see an Onboard GPU, and uncovered [KVMGT](https://01.org). After tinkering around for a few hours, I eventually was able to get the VM running Windows 10 to see and use the HOST's onboard GPU, from there sickbed_mp4 worked as usual. 

I have been testing this setup for the past 3 days and there have been no issues as of yet. 
I have also begun testing some of the ideas I mentioned earlier in this topic, such as deploying failover Virtualzed Plex/Emby Servers on the West coast and East coast, that have their media, metadata, watch status and etc... processed by the First/Original VM using sickbeard_mp4, and there has been no issues either with that portion of the setup also.
